modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_1_0/DDMFieldUpgradeProcess.java

```diff
@@ -96,65 +97,83 @@ protected void doUpgrade() throws Exception {
-						"DDMStructure.ctCollectionId = 0"));
-			PreparedStatement insertDDMFieldPreparedStatement =
-				AutoBatchPreparedStatementUtil.autoBatch(
-					connection,
-					StringBundler.concat(
-						"insert into DDMField (mvccVersion, ctCollectionId, ",
-						"fieldId, companyId, parentFieldId, storageId, ",
-						"structureVersionId, fieldName, fieldType, ",
-						"instanceId, localizable, priority) values (0, 0, ?, ",
-						"?, ?, ?, ?, ?, ?, ?, ?, ?)"));
-			PreparedStatement insertDDMFieldAttributePreparedStatement =
-				AutoBatchPreparedStatementUtil.autoBatch(
-					connection,
-					StringBundler.concat(
-						"insert into DDMFieldAttribute (mvccVersion, ",
-						"ctCollectionId, fieldAttributeId, companyId, ",
-						"fieldId, storageId, attributeName, languageId, ",
-						"largeAttributeValue, smallAttributeValue) values (0, ",
-						"0, ?, ?, ?, ?, ?, ?, ?, ?)"));
-			PreparedStatement deleteDDMContentPreparedStatement =
-				AutoBatchPreparedStatementUtil.autoBatch(
-					connection,
-					"delete from DDMContent where contentId = ? and " +
-						"ctCollectionId = 0");
-			ResultSet resultSet = selectPreparedStatement.executeQuery()) {
-
-			while (resultSet.next()) {
-				_upgradeDDMContent(
-					resultSet.getLong("companyId"),
-					resultSet.getLong("contentId"),
-					insertDDMFieldPreparedStatement,
-					insertDDMFieldAttributePreparedStatement,
-					resultSet.getString("data_"),
-					deleteDDMContentPreparedStatement,
-					resultSet.getLong("structureId"),
-					resultSet.getLong("structureVersionId"));
-			}
-
-			insertDDMFieldPreparedStatement.executeBatch();
-
-			insertDDMFieldAttributePreparedStatement.executeBatch();
-
-			deleteDDMContentPreparedStatement.executeBatch();
+		try (LoggingTimer loggingTimer = new LoggingTimer()) {
+			processConcurrently(
+				StringBundler.concat(
+					"select DDMContent.companyId, DDMContent.contentId, ",
+					"DDMContent.data_, DDMStorageLink.structureId, ",
+					"DDMStructure.structureVersionId from DDMContent inner ",
+					"join DDMStorageLink on DDMStorageLink.classPK = ",
+					"DDMContent.contentId inner join DDMStructureVersion on ",
+					"DDMStructureVersion.structureVersionId = ",
+					"DDMStorageLink.structureVersionId inner join ",
+					"DDMStructure on DDMStructureVersion.structureId = ",
+					"DDMStructure.structureId where ",
+					"DDMStructure.storageType = 'json' and ",
+					"DDMContent.ctCollectionId = 0 and ",
+					"DDMStorageLink.ctCollectionId = 0 and ",
+					"DDMStructureVersion.ctCollectionId = 0 and ",
+					"DDMStructure.ctCollectionId = 0"),
+				resultSet -> {
+					long companyId = resultSet.getLong(1);
+					long contentId = resultSet.getLong(2);
+					String data_ = resultSet.getString(3);
+					long structureId = resultSet.getLong(4);
+					long structureVersionId = resultSet.getLong(5);
+
+					return new Object[] {
+						contentId, companyId, data_, structureVersionId,
+						structureId
+					};
```

Comment:
Should be:
```java
return new Object[] {
    contentId, companyId, data_, structured,
    structureVersionId
};
```

Reply:
hi @mateussgn when I tried the upgrade process with this modification, it failed with the error
`com.liferay.portal.kernel.upgrade.UpgradeException: com.liferay.portal.kernel.upgrade.UpgradeException: Unable to find dynamic data mapping structure with ID 30850`
But without it, it was completed successfully.

Reply:
Hi @IstvanD, 
I've made some other changes after this one, can you try again?
After I made the changes I tested with an upgrade from 7.1 to master and was completed successfully.

Reply:
Hi @mateussgn thank you for the quick reply, I will test the changes. 👍 
I also got this error when upgrading the customer's database. Is this something that is maybe caused by the recent modifications?
```
2023-04-19 15:37:15.237 ERROR [main][UpgradeStepRegistratorTracker:185] Failed upgrade process for module com.liferay.dynamic.data.mapping.service
com.liferay.portal.kernel.upgrade.UpgradeException: java.sql.SQLSyntaxErrorException: Unknown column 'DDMStructure.structureVersionId' in 'field list'
```

Reply:
@mateussgn are your most recent changes in the commit [3e7d309](https://github.com/mateussgn/liferay-portal/pull/92/commits/3e7d309df87d3d7b82b8854e0e00138a14702fdb) that was pushed two days ago? Thank you and best regards.